drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -393,7 +396,22 @@ static int nrf5_tx(const struct device *dev,
 		nrf_802154_transmit_csma_ca_raw(nrf5_radio->tx_psdu);
 		break;
 	case IEEE802154_TX_MODE_TXTIME:
-	case IEEE802154_TX_MODE_TXTIME_CCA:
+	case IEEE802154_TX_MODE_TXTIME_CCA: {
+		bool cca = (mode == IEEE802154_TX_MODE_TXTIME_CCA);
+		uint32_t tx_at;
+
+		__ASSERT_NO_MSG(pkt);
+		__ASSERT(IS_ENABLED(CONFIG_NET_PKT_TIMESTAMP),
```

Comment:
Aren't we using wrong option here? There is the `CONFIG_NET_CONTEXT_TXTIME` option which is suppose to do this:
```
config NET_CONTEXT_TXTIME
	bool "Add TXTIME support to net_context"
	select NET_PKT_TXTIME
	help
	  It is possible to add information when the outgoing network packet
	  should be sent. The TX time information should be placed into
	  ancillary data field in sendmsg call.
```
Then in the user application it is possible to use `SO_TXTIME` socket option to select the exact tx time and the information is passed to the driver.

With your PR, who is setting the TX time? Is it the application?

Originally the `CONFIG_NET_PKT_TIMESTAMP` option was to be used by TSN time sync.

Reply:
I agree, it would be better to use `uint64_t txtime;` from net_pkt, but I'd rather stick to `CONFIG_NET_PKT_TXTIME` instead of `CONFIG_NET_CONTEXT_TXTIME` - we use the driver w/o IP layer as well (hence no net_context is used). In our case the upper layer will set the TX time (zigbee stack).

Reply:
Sure, if the is no sockets involved, then `CONFIG_NET_PKT_TXTIME` would be the correct symbol.

Reply:
I added `NET_PKT_TXTIME` symbol and a dedicated nrf symbol `IEEE802154_NRF5_PKT_TXTIME` that depends on `NET_PKT_TXTIME` according to https://github.com/zephyrproject-rtos/zephyr/pull/29123#discussion_r503257487